packages/grafana-data/src/types/datasource.ts

```diff
@@ -505,6 +505,16 @@ export interface DataQueryError {
   type?: DataQueryErrorType;
 }
 
+/**
+ * Describes the origin of the data request.
+ *
+ * @public
+ */
+export interface DataRequestOrigin {
+  type: 'panel' | 'variable';
```

Comment:
Some times queries are not associated with a single panel or variable, we actually have a lot more we want to track

Hm.. maybe something like this 

```ts
export interface DataRequestOrigin {
  queryType: 'panel-data' | 'variable'  | 'annotations' | 'shared';
  panelRef?: string;  // (panel id or could be panel name)
  queryRef?: string;  // Query name / ref for standalone queries, variable name for variable queries
  dashboardUID?: string;  
  pageName?:   // for scene apps plugins
  app?: string;   // the app plugin / or core feature name  (this already exists on the root DataQueryRequest)
}
```

Question what should we set for variable queries?

Reply:
I guess we also have alerting as a queryType? And probably more? 

The suggestion with having `type` and some kind of `reference` field was made so we could detect what kind of value is passed in the reference by reading the `type`.

I don't have as good knowledge of which of these values that can exist the same time on the same request, as you do, but it would be nice to end up with a generic structure that could support our current scenarios and hopefully most future scenarios without the need of adding new optional fields.

Reply:
Just a random comment from the past in regards to https://github.com/grafana/grafana/pull/11326. Back then already felt "dirty" that datasources have to populate origin information. I feel there's an opportunity to correlate the existing `requestId` with additional context/origin information allowing the  features, e.g. dashboard, explore, alerting, variables to provide the information that make sense to them. This can then be extracted in backend_srv given you have the `requestId`.